src/clean.rs

```diff
@@ -11,3 +13,18 @@ pub fn correct_h1(dom: &NodeRef) {
         }
     }
 }
+
+pub fn clean_html(code: String) -> Result<String, Box<dyn std::error::Error>> {
+    let client = reqwest::blocking::Client::new();
+
+    let res = client
+        .post("https://www.10bestdesign.com/dirtymarkup/api/html")
+        .form(&[("code", code), ("indent", "2".to_owned())])
```

Comment:
Hmm, does bikeshed really do this? Requiring internet to generate a bikeshed document or run tests is not great. Depending on this external API doesn't seem amazing.

Reply:
I'm trying to format the generated HTML and expected HTML with the same tool so that we can find the first line that doesn't match, but I couldn't find a Rust library for doing that.

Reply:
Where does bikeshed use this API? What is the package? Does this basically only take care of serializing the DOM in an indented way? If so, it may be better to just do it ourselves.

Reply:
`bikeshed` doesn't use that API, which basically just remove empty lines and serialize the DOM in an indented way. I use that API for string-based comparison.

If we want to use the test files of `bikeshed`, here're the options for us:

1. [DOM-based] Building DOM trees from files, and then comparing them without serialization.

    - It works now, but I'm not sure it will work later. Besides, it's hard to find errors in the result when we use this method.

2. [String-based] Writing a serializer exactly like the serializer of `bikeshed`.

3. [String-based] Building DOM trees from expected HTML files, and then serializing them back.

    - It doesn't work because of the indentation issue.

4. [String-based] Formatting the generated HTML and expected HTML with the same tool.

5. [DOM-based × String-based] Comparing DOM trees like method 1 first, and comparing strings like method 4 to find errors if the DOM trees are not equal.

Which one do you prefer?